MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -1,231 +1,152 @@
+  sqlite3 *db = NULL;
+  int result = SQLITE_OK;
+  result = sqlite3_open_v2([self.filePath UTF8String], &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL);
+  if (result == SQLITE_OK) {
+    char *errMsg;
+    result = sqlite3_exec(db, [query UTF8String], NULL, NULL, &errMsg);
+    if (result != SQLITE_OK) {
+      MSLogError([MSMobileCenter logTag], @"Query \"%@\" failed with error: %d - %@", query, result,
+                 [[NSString alloc] initWithUTF8String:errMsg]);
     }
+    sqlite3_close(db);
+  } else {
+    sqlite3_close(db);
+    MSLogError([MSMobileCenter logTag], @"Failed to open database.");
   }
-
-  // Get logs from DB that are not already part of a batch.
-  NSString *selectLogQuery =
-      [NSString stringWithFormat:@"SELECT * FROM %@ WHERE %@ == '%@' AND %@ NOT IN ('%@') LIMIT %lu", kMSLogTableName,
-                                 kMSGroupIdColumnName, groupId, kMSIdColumnName,
-                                 [idsInBatches componentsJoinedByString:@"','"], (unsigned long)limit];
-  return [self getLogsFromDBWithQuery:selectLogQuery];
+  return SQLITE_OK == result;
 }
 
-- (NSDictionary<NSString *, id<MSLog>> *)getLogsFromDBWithQuery:(NSString *)query {
-  NSArray<NSArray<NSString *> *> *result = [self.connection selectDataFromDB:query];
-  NSMutableDictionary<NSString *, id<MSLog>> *logs = [NSMutableDictionary<NSString *, id<MSLog>> new];
-
-  // Get logs from DB.
-  for (NSArray<NSString *> *row in result) {
-
-    // TODO use constants for DB column indexes.
-    NSString *Id = row[0];
-    NSString *base64Data = row[2];
-    NSData *logData =
-        [[NSData alloc] initWithBase64EncodedString:base64Data options:NSDataBase64DecodingIgnoreUnknownCharacters];
-    id<MSLog> log;
-
-    // Deserialize the log.
-    @try {
-      log = [NSKeyedUnarchiver unarchiveObjectWithData:logData];
-    } @catch (NSException *exception) {
-
-      // The archived log is not valid.
-      MSLogError([MSMobileCenter logTag], @"Deserialization failed for log with Id %@: %@", Id, exception);
-      [self deleteLogFromDBWithId:Id];
-      continue;
+- (NSArray<NSArray *> *)executeSelectionQuery:(NSString *)query {
+  NSMutableArray<NSMutableArray *> *entries = [[NSMutableArray<NSMutableArray *> alloc] init];
+  sqlite3 *db = NULL;
+  sqlite3_stmt *statement = NULL;
+  int result = 0;
+  result = sqlite3_open_v2([self.filePath UTF8String], &db, SQLITE_OPEN_READONLY, NULL);
+  if (result == SQLITE_OK) {
+    result = sqlite3_prepare_v2(db, [query UTF8String], -1, &statement, NULL);
+    if (result == SQLITE_OK) {
+      // Loop on rows.
+      while (sqlite3_step(statement) == SQLITE_ROW) {
+        NSMutableArray *entry = [NSMutableArray new];
+
+        // Loop on columns.
+        for (int i = 0; i < sqlite3_column_count(statement); i++) {
+          id value = nil;
+
+          // Convert values.
+          switch (sqlite3_column_type(statement, i)) {
+          case SQLITE_INTEGER:
```

Comment:
I know I had the same thoughts. But then I'll have to add more types also on the schema, I don't think we need to do this work now.